dagjose/multicodec.go

```diff
@@ -1,11 +1,37 @@
 package dagjose
 
 import (
-	dagcbor "github.com/ipld/go-ipld-prime/codec/dagcbor"
-	multicodec "github.com/ipld/go-ipld-prime/multicodec"
+	"io"
+
+	"github.com/ipld/go-ipld-prime/codec/dagcbor"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	"github.com/ipld/go-ipld-prime/multicodec"
 )
 
 func init() {
-	multicodec.RegisterDecoder(0x85, dagcbor.Decode)
-	multicodec.RegisterEncoder(0x85, dagcbor.Encode)
+	multicodec.RegisterDecoder(0x85, Decode)
+	multicodec.RegisterEncoder(0x85, Encode)
+}
+
+// Decode deserializes data from the given io.Reader and feeds it into the
+// given datamodel.NodeAssembler. Decode fits the codec.Decoder function
+// interface.
+func Decode(na datamodel.NodeAssembler, r io.Reader) error {
+	return dagcbor.Decode(na, r)
```

Comment:
I got the CBOR encoding to work fine in my repo. I'll make a PR with some of the cherry-picked features excluding stuff like renaming the `go.mod`.

Reply:
Nice! Thanks for checking (and sorry it took so long to get here 😅).

The reason for the `datamodel.Copy` is not because the CBOR encoding won't work without it, it is to make sure that only _valid_ JOSE gets CBOR encoded.

Without this change I'm able to store any valid CBOR object even if it is _invalid_ JOSE.